### PR TITLE
Remove `ignore-compare-mode-next-solver` from tests that don't hang anymore

### PR DESCRIPTION
**Pull request by compiler-errors**
self-explanatory

r? lcnr or anyone really

### CODE REVIEW

File: tests/ui/iterators/issue-58952-filter-type-length.rs [hunk 1]
```diff
@@ -1,5 +1,4 @@
 //@ run-pass
-//@ ignore-compare-mode-next-solver (hangs)
```
Comment (compiler-errors) [inline tests/ui/iterators/issue-58952-filter-type-length.rs:2]: Actually this one still hangs 🤦

**Comment by lqd** [comment]
r=me after blessing

**Comment by compiler-errors** [comment]
Right, I forgot that the line numbers will change :|

**Comment by compiler-errors** [comment]
gotta do this from scratch

**Pull request closed without merging**